### DISCUSSION TITLE
Additional black stripes while concat images by vips arrayjoin

### DISCUSSION THREAD

**Discussion by joesong168** [libvips/libvips#3593] [Q&A]
## Questions, enhancements, tips, etc.
As shown in the image, there happened to be several black stripes in the generated image by vips arrayjoin. I tried to remove them by setting shims to 0 with no luck. Is this a bug or a feature?
![屏幕截图 2023-08-03 225619](https://github.com/libvips/libvips/assets/10493864/91d7fb38-d4b1-4c1b-8e64-f413b3e66966)

Regards,
Joe

**Comment by jcupitt** [comment]
Hi @joesong168,

`arrayjoin` assembles images in a regular grid, so the images must all be the same size of you'll get gaps.

If your images differ in width, you need `join` (for hard joins) or `merge` (for a soft join with an overlap).

**Reply by joesong168** [reply]
Thanks for the answer. So does hard joins means the code/decode of the image while soft join leaves coding untouched?
